lib/srv/desktop/tdp/proto.go

```diff
@@ -513,14 +550,21 @@ func (c ClipboardData) Encode() ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+const clipDataMaxLenErr = "clipboard sync failed: clipboard data exceeded maximum length"
+
 func decodeClipboardData(in io.Reader, maxLen uint32) (ClipboardData, error) {
 	var length uint32
 	if err := binary.Read(in, binary.BigEndian, &length); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
 	if length > maxLen {
-		return nil, trace.BadParameter("clipboard data exceeds maximum length")
+		// If clipboard data exceeds maxLen,
+		// discard the rest of the message
+		if _, err := io.CopyN(io.Discard, in, int64(length)); err != nil {
```

Comment:
Do we need this check? Can we just skip straight to returning the max length error? If we don't, and someone sends a clipboard data message with a length field that is too large, but they don't actually send that much data, we'll return an EOF error instead of the max data error.

Additionally, would `trace.LimitExceeded` be better than `BadParameter` here?